system.nix

```diff
@@ -62,6 +62,10 @@ in
   config = {
     boot.supportedFilesystems = [ "zfs" ];
     boot.initrd.postDeviceCommands = "${post-device-cmds}";
+    boot.initrd.preDeviceCommands = ''
+      echo "Interrupt within 3 seconds to get a shell..."
+      sleep 3 || exec sh
```

Comment:
should this be, like, --interactive or something? is `sh` in this case `ash`? I think probably so. Might double check this, and also double check if you have the tools you need in PATH to do something useful.